### PR TITLE
Fix XNNPACK Resize with empty scales

### LINKED ISSUE #32298: XNNPACK EP on Android arm64 aborts the process (Abort message: 'terminating') instead of returning an error — Resize with empty scales, still in 1.26.0
### Describe the issue

Creating an `InferenceSession` with the XNNPACK EP on Android arm64 **aborts the whole process** (`SIGABRT`, `Abort message: 'terminating'`) instead of returning an error. Because the C++ exception escapes through `std::terminate`, a managed `try/catch` around session creation cannot intercept it, so an application has no way to probe for XNNPACK support and fall back to CPU.

This is the same failure mode reported in #23826 (closed as a documentation issue). That reporter tracked the root cause down to `Resize` nodes whose `scales` input is an **empty tensor**, and worked around it by patching their own copy of the XNNPACK wrapper. The underlying crash is still present in **1.26.0**, and it reproduces with a widely used public model.

Two separable problems:

1. XNNPACK cannot handle `Resize` with an empty `roi`/`scales` initializer (opset 11 style, produced by `tf2onnx`).
2. Regardless of (1) — an unsupported node should not terminate the process. Session creation should fail with a catchable status so callers can fall back to the CPU EP.

### To reproduce

**Model:** MoveNet SinglePose Lightning v4, `tf2onnx` conversion — public and directly downloadable:

```
https://raw.githubusercontent.com/Kazuhito00/MoveNet-Python-Example/515743a113e49b4f31677b1f6252c0041f237e30/onnx/movenet_singlepose_lightning_4.onnx
```

Input `1x192x192x3` int32 NHWC, output `1x1x17x3`, `opset_import: ai.onnx v11`.

The model contains three `Resize` nodes, each with `roi` and `scales` bound to the same zero-length initializer:

```
Resize__347  inputs = [ ...conv2d/BiasAdd:0, const_empty_float__341, const_empty_float__341, Concat__346:0 ]
Resize__368  inputs = [ ...re_lu/Relu:0,     const_empty_float__341, const_empty_float__341, Concat__367:0 ]
Resize__389  inputs = [ ...re_lu_1/Relu:0,   const_empty_float__341, const_empty_float__341, Concat__388:0 ]

const_empty_float__341: initializer, dims = [0]   # empty tensor
```

**Code** (C#, `Microsoft.ML.OnnxRuntime` 1.26.0, .NET 10 Android):

```csharp
using var options = new SessionOptions();
options.AppendExecutionProvider(
    "XNNPACK",
    new Dictionary<string, string> { ["intra_op_num_threads"] = "2" });

// Never returns. The process aborts here; the surrounding try/catch never runs.
using var session = new InferenceSession(modelBytes, options);
```

The same model loads and runs correctly with default session options (CPU EP) on the same device.

### Urgency

Not blocking — we fall back to the CPU EP. But it costs us the ability to even *probe* for XNNPACK at runtime: a process-level abort cannot be guarded, so the only safe option is never to request the EP at all.

### Platform

Android

### OS Version

Android 15 (`realme/RMX3771RU/RE58B8L1:15/AP3A.240617.008`), arm64-v8a, MediaTek Dimensity 7050

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.26.0 (`Microsoft.ML.OnnxRuntime` NuGet; native library from `runtimes/android/native/onnxruntime.aar`, `jni/arm64-v8a/libonnxruntime.so`)

### ONNX Runtime API

C#

### Architecture

ARM64

### Execution Provider

Other / Unknown (XNNPACK)

### Execution Provider Library Version

XNNPACK as shipped inside the 1.26.0 Android AAR (`XnnpackExecutionProvider` symbols are present in `jni/arm64-v8a/libonnxruntime.so`, so the EP is compiled in).

### Actual behaviour

```
F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 32022 (.NET TP Worker)
F DEBUG   : Build fingerprint: 'realme/RMX3771RU/RE58B8L1:15/AP3A.240617.008/T.231124a_d5daf:user/release-keys'
F DEBUG   : ABI: 'arm64'
F DEBUG   : pid: 31996, tid: 32022, name: .NET TP Worker
F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
F DEBUG   : Abort message: 'terminating'
F DEBUG   : backtrace:
F DEBUG   :       #00 pc 0000000000093dc8  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168)
F DEBUG   :       #01 pc 00000000009c32a8  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #02 pc 00000000009c2d04  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #03 pc 00000000009c2ca0  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #04 pc 00000000009c2c64  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #05 pc 00000000009ff2a8  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #06 pc 0000000000b6ace4  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #07 pc 0000000000b79f04  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #08 pc 0000000000b53d94  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #09 pc 00000000013d810c  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #10 pc 00000000013dad94  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #11 pc 00000000013d85d4  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #12 pc 00000000013d49d8  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #13 pc 0000000000a2223c  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #14 pc 0000000000a26d60  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #15 pc 0000000000acdd5c  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #16 pc 00000000009e89f4  <app>/lib/arm64/libonnxruntime.so
F DEBUG   :       #17 pc 00000000001abc08  <app>/lib/arm64/libmonosgen-2.0.so
...
F DEBUG   :       #21 pc 000000000025acc8  <app>/lib/arm64/libmonosgen-2.0.so (mono_runtime_invoke_checked+140)
```

(`libonnxruntime.so` BuildId `de83f9cb47b2efdf4d6ddb222f4ac169dec0542e`. The library is unstripped only to the extent shipped in the AAR, hence the bare addresses.)

### Expected behaviour

Session creation with an unsupported model should fail with an `OrtStatus` / `OnnxRuntimeException` that the caller can catch and handle — for example by falling back to the CPU EP — rather than calling `std::terminate` and taking the process down.

Additionally, `Resize` with an empty `roi`/`scales` initializer is valid ONNX (both inputs are optional and this is what `tf2onnx` emits for opset 11); XNNPACK should either handle it or decline to take the node during partitioning.

### PR DESCRIPTION
**Pull request by Jokasa7**
### Description

- Treat zero-element `scales` and `sizes` initializers as unused Resize inputs before inspecting their values.
- Validate rank-4 geometry inputs and the inferred output shape before indexing dimensions.
- Add an XNNPACK regression test for Resize with empty `scales` and a populated `sizes` input.

### Motivation and Context

ONNX Resize models can represent the unused `scales` input as a zero-element initializer when `sizes` supplies the output geometry. The XNNPACK capability check treated that initializer as populated and indexed its empty span, aborting session initialization.

Fixes #32298.

Local validation:

- Built `onnxruntime_provider_test` with XNNPACK enabled.
- Before the fix, the focused regression ran 1 test and aborted with exit code 6.
- After the fix, `XnnpackEP.TestResizeEmptyScalesUsesSizes` passed (1 test).
- `XnnpackEP.*Resize*` passed (2 tests).
- `XnnpackEP.*` passed (20 tests).
- `clang-format 19.1.5 --dry-run --Werror --style=file` passed for both changed files.
- `git diff --check` passed.

### CODE REVIEW

File: onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc [hunk 1]
```diff
@@ -521,6 +521,26 @@ TEST(XnnpackEP, Resize) {
   RunModelTestWithPath(ort_model_path, "test_resize", nullptr);
 }
 
+TEST(XnnpackEP, TestResizeEmptyScalesUsesSizes) {
+  auto model_builder = [](ModelTestBuilder& builder) {
+    auto* input = builder.MakeInput<float>({1, 3, 8, 8}, -1.0f, 1.0f);
+    auto* roi = builder.MakeInitializer<float>({0}, {});
+    auto* scales = builder.MakeInitializer<float>({0}, {});
+    auto* sizes = builder.MakeInitializer<int64_t>({4}, {1, 3, 16, 16});
```
[the remark this answers is not in the document]
Comment (Jokasa7) [reply onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc:529]: Thanks. I checked this against the ONNX schema and ORT's CPU Resize contract. A zero-element `sizes` tensor is not the symmetric omission form: ONNX represents an unused optional input by omitting it, and ORT's CPU kernel rejects populated `scales` plus any `sizes` tensor with `Only one of scales or sizes must be provided as input.` Commit 7799323 now explicitly rejects empty or non-rank-sized `sizes` in the XNNPACK capability check instead of treating it as unused. The valid empty-`scales` + populated-`sizes` regression remains covered. The final Release build passed, including Resize 3/3 and XnnpackEP 21/21.

File: onnxruntime/core/providers/xnnpack/tensor/resize.cc [hunk 2]
```diff
@@ -57,46 +57,67 @@ bool Resize::IsOnnxNodeSupported(const NodeUnit& node_unit,
...
 
+    // Exactly one input must provide the output geometry. A zero-element tensor represents an unused input.
+    if (has_scales_data == has_sizes_data) {
+      break;
+    }
+
     const auto* output_shape = node_unit.Outputs()[0].node_arg.Shape();
+    if (!output_shape || output_shape->dim_size() != 4) {
+      break;
+    }
```
[the remark this answers is not in the document]
Comment (Jokasa7) [reply onnxruntime/core/providers/xnnpack/tensor/resize.cc:121]: Fixed in 7799323 by rejecting unknown or non-positive spatial dimensions before XNNPACK assignment. I also added `TestResizeDynamicSpatialDimensionsFallBack`, which uses symbolic H/W with constant `sizes`, provides a concrete runtime feed, and verifies CPU fallback instead of entering the XNNPACK constructor with an empty `output_dims_`. The final Release build passed; focused tests were 2/2, Resize tests 3/3, and XnnpackEP 21/21.
